### PR TITLE
Improve C++ highlighter

### PR DESCRIPTION
**Pull request by data-man**
Added missed C++11/20 keywords and some types.

### CODE REVIEW

**Comment by pbek** [comment]
@Waqar144, fine with you?

**Comment by Waqar144** [comment]
All good. We need to get rid of these globals though. I am working on a plan

File: qownlanguagedata.cpp [hunk 1]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
+        {('a'), QLatin1String("and")},
+        {('a'), QLatin1String("and_eq")},
         {('a'), QLatin1String("asm")},
         {('a'), QLatin1String("auto")},
+        {('b'), QLatin1String("bit_and")},
+        {('b'), QLatin1String("bit_or")},
         {('b'), QLatin1String("break")},
         {('c'), QLatin1String("case")},
         {('c'), QLatin1String("catch")},
+        {('c'), QLatin1String("class")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:70]: why class is moved here?

File: qownlanguagedata.cpp [hunk 2]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
+        {('c'), QLatin1String("co_await")},
+        {('c'), QLatin1String("co_return")},
+        {('c'), QLatin1String("co_yield")},
+        {('d'), QLatin1String("decltype")},
         {('d'), QLatin1String("default")},
         {('d'), QLatin1String("delete")},
         {('d'), QLatin1String("do")},
         {('d'), QLatin1String("dynamic_cast")},
         {('e'), QLatin1String("else")},
+        {('e'), QLatin1String("enum")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:88]: why `enum` is moved here from `cpp_types`?

File: qownlanguagedata.cpp [hunk 3]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
+        {('r'), QLatin1String("requires")},
         {('r'), QLatin1String("return")},
         {('s'), QLatin1String("signal")},
         {('s'), QLatin1String("signed")},
         {('s'), QLatin1String("sizeof")},
         {('s'), QLatin1String("slot")},
         {('s'), QLatin1String("static")},
+        {('s'), QLatin1String("static_assert")},
         {('s'), QLatin1String("static_cast")},
+        {('s'), QLatin1String("struct")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:120]: why `struct` is moved here from `cpp_types`?

File: qownlanguagedata.cpp [hunk 4]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
+        {('t'), QLatin1String("thread_local")},
         {('t'), QLatin1String("throw")},
         {('t'), QLatin1String("try")},
+        {('t'), QLatin1String("typeid")},
         {('t'), QLatin1String("typedef")},
+        {('t'), QLatin1String("typename")},
         {('u'), QLatin1String("unsigned")},
+        {('u'), QLatin1String("union")},
         {('u'), QLatin1String("using")},
+        {('v'), QLatin1String("void")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:133]: why `void` is moved here from `cpp_types`?

File: qownlanguagedata.cpp [hunk 5]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
         {('s'), QLatin1String("switch")},
         {('t'), QLatin1String("template")},
         {('t'), QLatin1String("this")},
+        {('t'), QLatin1String("thread_local")},
         {('t'), QLatin1String("throw")},
         {('t'), QLatin1String("try")},
+        {('t'), QLatin1String("typeid")},
         {('t'), QLatin1String("typedef")},
+        {('t'), QLatin1String("typename")},
         {('u'), QLatin1String("unsigned")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:130]: this should also go into `cpp_types`

File: qownlanguagedata.cpp [hunk 6]
```diff
@@ -56,63 +56,129 @@ static QMultiHash<char, QLatin1String> cpp_literals;
...
+        {('u'), QLatin1String("union")},
```
Comment (Waqar144) [inline qownlanguagedata.cpp:131]: why `union` is moved here from cpp_types?

**Comment by Waqar144** [comment]
We need to get rid of these globals though.

**Comment by data-man** [comment]
> why class is moved here?
> why enum is moved here from cpp_types?
> why struct is moved here from cpp_types?
> why void is moved here from cpp_types?
> this should also go into cpp_types
> why union is moved here from cpp_types?

https://en.cppreference.com/w/cpp/keyword

**Comment by Waqar144** [comment]
> > why class is moved here?
> > why enum is moved here from cpp_types?
> > why struct is moved here from cpp_types?
> > why void is moved here from cpp_types?
> > this should also go into cpp_types
> > why union is moved here from cpp_types?
> 
> https://en.cppreference.com/w/cpp/keyword

?
That is just a list of keywords. 'types' go in types. anything that makes a type, goes in types. So this needs to be fixed.

**Pull request merged**